esmvalcore/_recipe.py

```diff
@@ -665,13 +696,21 @@ def get_matching(attributes):
 
 
 def _get_preprocessor_products(variables, profile, order, ancestor_products,
-                               config_user):
+                               config_user, fx_case=False):
```

Comment:
it is a flag that determines the coupling between the main`var` or `variable` and the fx variable - a kludge (or hack) that's needed to let the workflow know that it needs info from both and pass it to the data finder and juggle the task creating correctly. This will disappear once we remove the ancestry for fx vars preprocessing (I renamed it to `var_fxvar_coupling` for now to make it more relevant)